analyses/focal-cn-file-preparation/06-find-recurrent-calls.Rmd

```diff
@@ -0,0 +1,189 @@
+  group_by(status, region, region_type) %>%
+  tally(name = "count") %>%
+  # Add column named `short_histology` for joining later
+  mutate(short_histology = "All")
+
+# Display table with counts of each unique status call by region
+final_df_tally %>%
+  arrange(desc(count))
+```
+
+### Tally the sample counts of each unique status call by region and short histology.
+
+```{r}
+# Tally the number of samples that have the same status call by region
+final_df_tally_histology <- focal_cn_df %>%
+  # Join the metadata by biospecimen IDs to get short histology data
+  inner_join(select(metadata,
+                    c(
+                      "Kids_First_Biospecimen_ID", "short_histology"
+                    )),
+             by = "Kids_First_Biospecimen_ID") %>%
+  group_by(status, region, region_type, short_histology) %>%
+  tally(name = "count")
+
+# Display table with counts of each unique status call by region and short
+# histology
+final_df_tally_histology %>%
+  arrange(desc(count))
+```
+
+### Plot the distribution of counts
+
+```{r}
+# Plot the distribution of counts for all histologies
+ggplot(final_df_tally, aes(x = count)) +
+  geom_histogram(bins = 100) +
+  geom_vline(aes(xintercept = recurrence_cutoff), linetype = "longdash", color = "green", size = .4)
+```
+
+```{r}
+# Plot the distribution of counts for histologies separated
+ggplot(final_df_tally_histology, aes(x = count)) +
+  geom_histogram(bins = 100) +
+  geom_vline(aes(xintercept = recurrence_cutoff), linetype = "longdash", color = "green", size = .4)
+```
+
+The `recurrence_cutoff` defined as 3 appears to be reasonable as seen in the distribution of counts plot above.
+
+### Isolate recurrent calls
+
+#### All histologies represented
+
+```{r}
+# Create a data frame of only observations that have counts greater than the `recurrence_cutoff`
+recurrent_df <- final_df_tally %>%
+  filter(count > recurrence_cutoff)
+
+# Display table
+recurrent_df
+```
+
+#### Separated by histology
+
+```{r}
+# Create a data frame of only observations that have counts greater than the `recurrence_cutoff`
+recurrent_df_histology <- final_df_tally_histology %>%
+  filter(count > recurrence_cutoff)
+
+# Display table
+recurrent_df_histology
+```
+
+#### Combine all histology recurrent data with the data separated by histology 
+
+```{r}
+final_df_recurrent <- recurrent_df %>%
+  bind_rows(recurrent_df_histology) %>%
+  arrange(region) %>%
+  select(region, status, short_histology, count, region_type)
+
+# Display table
+final_df_recurrent
+```
+
+### Save list of genes that have `recurrent` calls
+
+```{r}
+# Save list of genes that have `recurrent` status calls for use in downstream
+# analyses
+top_recurrent_genes <- final_df_recurrent %>%
+  # Ungroup data from previous step
+  ungroup() %>%
+  # Arrange by descending count
+  arrange(desc(count)) %>%
+  # Filter to only gene regions
+  filter(region_type == "gene") %>%
+  # Select the top genes by count
+  top_n(20, count) %>%
```

Comment:
@jaclyn-taroni if I move this step before filtering to only gene regions, it will return the top `n` for all regions. Can you further clarify what you mean here?

Reply:
The number 20 is hardcoded here, assign 20 to a variable up top.

Reply:
Ah gotcha 👍